cmd/metal-api/internal/service/machine-service.go

```diff
@@ -440,6 +454,47 @@ func (r machineResource) findMachine(request *restful.Request, response *restful
 	}
 }
 
+func (r machineResource) getMachineConsolePassword(request *restful.Request, response *restful.Response) {
+	var requestPayload v1.MachineConsolePasswordRequest
+	err := request.ReadEntity(&requestPayload)
+	if checkError(request, response, utils.CurrentFuncName(), err) {
+		return
+	}
+
+	if len(requestPayload.Reason) < 5 {
+		if checkError(request, response, utils.CurrentFuncName(), fmt.Errorf("reason must be at least 5 characters long")) {
+			return
+		}
+	}
+	user, err := r.userGetter.User(request.Request)
+	if checkError(request, response, utils.CurrentFuncName(), err) {
+		return
+	}
+
+	m, err := r.ds.FindMachineByID(requestPayload.ID)
+	if checkError(request, response, utils.CurrentFuncName(), err) {
+		return
+	}
+
+	consolePassword := ""
+	if m.Allocation != nil && m.Allocation.ConsolePassword != "" {
```

Comment:
We should return an error when requesting a console password for an unallocated machine, otherwise a user retrieves `password:` as a result, which can be misleading.

Reply:
fixed